Sources/MapboxCoreNavigation/HistoryRecording.swift

```diff
@@ -102,9 +102,7 @@ public extension HistoryRecording {
     }
 
     static func startRecordingHistoryImplementation() {
-        if Navigator.isSharedInstanceCreated {
-            Navigator.shared.historyRecorder?.startRecording()
-        }
+        HistoryRecorder.shared.handle?.startRecording()
```

Comment:
Static API is deprecated. Should we now deprecate instance API and undeprecate static one? Or is it better to create a standalone (basically empty) `HistoryRecorder` public class that implements `HistoryRecording`?

Reply:
Exposing a dedicated (singleton) `HistoryRecorder` defeats the purpose of `HistoryRecording` protocol at which point we should deprecate it entirely. From one perspective this sounds reasonable, so that users see that this is a separate functionality with it's own lifecycle, but on the other - it seems like an overcomplication since we already have `HistoryRecording` available which capable of doing the same thing (given we undeprecate static methods)